src/lib.rs

```diff
@@ -8,22 +8,37 @@
 
 #![doc(html_root_url = "https://docs.rs/itoa/0.3.4")]
 
+#![cfg_attr(not(feature = "use_std"), no_std)]
+
 #![cfg_attr(feature = "i128", feature(i128_type, i128))]
 
 #![cfg_attr(feature = "cargo-clippy", allow(cast_lossless, unreadable_literal))]
 
 #[cfg(feature = "i128")]
 mod udiv128;
 
-use std::{io, mem, ptr, slice};
+#[cfg(feature = "use_std")]
+use std::{fmt, io, mem, ptr, slice, str};
+
+#[cfg(not(feature = "use_std"))]
+use core::{fmt, mem, ptr, slice, str};
 
+#[cfg(feature = "use_std")]
 #[inline]
 pub fn write<W: io::Write, V: Integer>(wr: W, value: V) -> io::Result<usize> {
     value.write(wr)
 }
 
+#[inline]
+pub fn fmt<W: fmt::Write, V: Integer>(wr: W, value: V) -> fmt::Result {
+    value.fmt(wr)
+}
+
 pub trait Integer {
+    #[cfg(feature = "use_std")]
     fn write<W: io::Write>(self, W) -> io::Result<usize>;
```

Comment:
I think this is still problematic.

> Cargo features are required to be additive, i.e. any code that compiles without the feature must also compile with the feature.

In this case if someone downstream writes a type:

```rust
struct u24(/* ... */);

impl Integer for u24 {
    fn fmt<W: fmt::Write>(self, W) -> fmt::Result {
        /* ... */
    }
}
```

their code would compile fine when our std feature is not enabled but would break when some unrelated crate depends on our std feature. That makes the feature not additive.

See [C-SEALED](https://rust-lang-nursery.github.io/api-guidelines/future-proofing.html#c-sealed) for a pattern that would work here.